ecosystem/typescript/sdk/src/hex_string.ts

```diff
@@ -0,0 +1,53 @@
+import { Buffer } from "buffer/"; // the trailing slash is important!
+import Types from "./types";
+
+// eslint-disable-next-line no-use-before-define
+export type MaybeHexString = HexString | string | Types.HexEncodedBytes;
+
+export class HexString {
+  /// We want to make sure this hexString has the `0x` hex prefix
+  private readonly hexString: string;
+
+  static fromBuffer(buffer: Buffer): HexString {
+    return new HexString(buffer.toString("hex"));
+  }
+
+  static fromUint8Array(arr: Uint8Array): HexString {
+    return HexString.fromBuffer(Buffer.from(arr));
+  }
+
+  static ensure(hexString: MaybeHexString): HexString {
+    if (typeof hexString === "string") {
+      return new HexString(hexString);
+    }
+    return hexString;
+  }
+
+  constructor(hexString: string | Types.HexEncodedBytes) {
+    if (hexString.startsWith("0x")) {
+      this.hexString = hexString;
+    } else {
+      this.hexString = `0x${hexString}`;
```

Comment:
should this log a warning? or at least something to signify a non-standard reading?

Reply:
I don't know if we are ready for that: we should standardize to _always_ return and accept 0x strings, as it's rather haphazard

Reply:
agreed on the the 0x prefixing. I guess the one thing that would be really great is if we could somehow infer that a 0x meant it was hex encoded where as no prefix implied it was a ascii / unicode string. Alternatively we can just make APIs with light typing that make it explicit, like you have done -- but that doesn't work for typeless languages nor web endpoints :(